pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -88,9 +88,10 @@ func (configgen *ConfigGeneratorImpl) buildSidecarInboundHTTPRouteConfig(
 	defaultRoute := istio_route.BuildDefaultHTTPInboundRoute(node, clusterName, traceOperation)
 
 	inboundVHost := &route.VirtualHost{
-		Name:    fmt.Sprintf("%s|http|%d", model.TrafficDirectionInbound, instance.ServicePort.Port),
-		Domains: []string{"*"},
-		Routes:  []*route.Route{defaultRoute},
+		Name:                       fmt.Sprintf("%s|http|%d", model.TrafficDirectionInbound, instance.ServicePort.Port),
+		Domains:                    []string{"*"},
+		Routes:                     []*route.Route{defaultRoute},
+		IncludeRequestAttemptCount: true,
```

Comment:
As retry happens at upstream, not sure if the inbound setting will override the incoming headers.

Reply:
That is a good point and also we do not need it for inbound routes. So removed it.

Reply:
ok